src/main/daml/Daml/Finance/Interface/Instrument/Swap/Asset/Types.daml

```diff
@@ -0,0 +1,37 @@
+-- Copyright (c) 2022 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+module Daml.Finance.Interface.Instrument.Swap.Asset.Types where
+
+import Daml.Finance.Interface.Instrument.Base.Instrument qualified as BaseInstrument (K)
+import Daml.Finance.Interface.Types.Common (InstrumentKey(..), PartiesMap)
+import Daml.Finance.Interface.Types.Date.DayCount (DayCountConventionEnum)
+import Daml.Finance.Interface.Types.Date.Schedule (PeriodicSchedule(..))
+
+data Asset = Asset
+  with
+    instrumentKey : InstrumentKey
+      -- ^ The instrument's key.
+    description : Text
+      -- ^ The description of the swap.
+    referenceAssetId : Text
```

Comment:
```suggestion
    referenceAssetId : Id
```
Or should this been a `InstrumentKey` ?

Reply:
It's used as an `Observable` in the CC tree, so has to be `Text` - the thing being observed here could be a non-instrument (eg. an interest rate). We could consider calling it `referenceObservable` but that is - while technically more accurate - probably more confusing to a user than helping.